test/jdk/java/lang/Thread/SleepSanity.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @summary Sanity test Thread.sleep behavior
+ * @run junit SleepSanity
+ */
+
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class SleepSanity {
+
+    static final int[] TRY_MILLIS = new int[] { 0, 1, 10, 100, 1_000 };
+    static final int[] TRY_NANOS  = new int[] { 0, 1, 10, 100, 1_000, 10_000, 100_000, 999_999 };
+
+    @Test
+    void testMs() throws Exception {
+        testIAE(() -> Thread.sleep(-1), "timeout value is negative");
+
+        testTimeout(() -> Thread.sleep(10_000),            5_000);
+        testTimeout(() -> Thread.sleep(Integer.MAX_VALUE), 5_000);
+        testTimeout(() -> Thread.sleep(Long.MAX_VALUE),    5_000);
+
+        for (final int millis : TRY_MILLIS) {
+            testTimes(() -> Thread.sleep(millis), millis, 10_000);
```

Comment:
I wonder if 10s is enough a tolerance. JDK-8303633 has some sightings of sleep(1000) taking 5.5s. We've since changed all of these to 20s. It might be that 10s is okay, it's just that we seem to run on some test systems (Windows mostly) where we need very high tolerance for tests like this.

Reply:
Yeah, I was wondering about the same when writing the test. I have not seen it failing yet, and it was pretty slow without the parallelization. To handle the ~5s hiccups, I bumped all "times" tests from 10s to 20s. I kept "timeout" tests at the same 5s timeout, because they normally do wait for those 5s. The accidental hiccup would make timeout test accidentally pass, which I think is a fair trade for keeping the test execution times low.